include/clover_field.h

```diff
@@ -213,12 +214,18 @@ namespace quda {
 
   public:
     CloverField(const CloverFieldParam &param);
-    virtual ~CloverField();
 
     static CloverField *Create(const CloverFieldParam &param);
 
-    void* V(bool inverse=false) { return inverse ? cloverInv : clover; }
-    const void* V(bool inverse=false) const { return inverse ? cloverInv : clover; }
+    template <typename T = void *> auto data(bool inverse = false) const
+    {
+      return inverse ? reinterpret_cast<T>(cloverInv.data()) : reinterpret_cast<T>(clover.data());
```

Comment:
Pedantic/in part for my own comprehension of how you implemented `.data()`---could this also be implemented as:

```
return inverse ? cloverInv.data<T*>() : clover.data<T*>();
```

I know it'd all be a means to the same end, I figured I'd ask for my own understanding.

Reply:
No, because the underlying type here is a `quda_ptr` which is not templated.